torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -27,7 +27,14 @@ const std::string kServerActiveAsyncCalls = "agent.server_active_async_calls";
 const std::string kRpcTimeoutErrorStr =
     "RPC ran for more than set timeout ({} ms) and will now be marked with an error";
 
-std::string guessAddress(tensorpipe::transport::uv::Context& uvContext) {
+} // namespace
+
+C10_DEFINE_REGISTRY(TensorPipeTransportRegistry, TransportRegistration);
```

Comment:
What does `C10_DEFINE_REGISTRY`/`C10_DECLARE_REGISTRY ` Do? Shall we have some comments here?

Reply:
They are defined in https://github.com/pytorch/pytorch/blob/master/c10/util/Registry.h and there's already some comments in that file... do you think there's any more info we could add here without repeating ourselves?

In a nutshell, a registry is a string-to-generator map. C10_DECLARE_REGISTRY adds the _declaration_ of a registry, i.e., the part that must go in headers. C10_DEFINE_REGISTRY adds the _definition_, i.e., the part that must go in source files.

Reply:
I see. If a 3rd-party user would like to register a new transport/channel, will they go through the same or a different process?

Reply:
At the moment we're not prioritizing that specific use case, which means that, yes, for now to add a new transport/channel users need to write a file where they include `C10_REGISTER_CREATOR(TensorPipeChannelRegistry, their_name, ...)` and then link it together with PyTorch. It is indeed quite clunky. With c10d::Extension we would be able to provide a second path to register transport/channels, which would be more user-friendly.

Reply:
Note that also the API of transports/channels isn't really finalized (we're not making explicit properties about its stability, it should be considered private), so users might incur other problems if they were to trying to plug in their custom channel.